e2e/testcases/workload_identity_test.go

```diff
@@ -53,168 +63,234 @@ import (
 // 5. The source of truth is hosted on a GCP service, e.g. CSR, GAR, GCR.
 func TestWorkloadIdentity(t *testing.T) {
 	testCases := []struct {
-		name                        string
-		fleetWITest                 bool
-		crossProject                bool
-		sourceRepo                  string
-		sourcePackage               string
-		sourceChart                 string
-		sourceVersion               string
-		sourceType                  v1beta1.SourceType
-		gsaEmail                    string
-		rootCommitFn                nomostest.Sha1Func
-		testKSAMigration            bool
-		requireHelmArtifactRegistry bool
-		requireOCIArtifactRegistry  bool
-		requireCSR                  bool
+		name             string
+		fleetWITest      bool
+		crossProject     bool
+		rootSrcCfg       sourceConfig
+		nsSrcCfg         sourceConfig
+		sourceType       v1beta1.SourceType
+		gsaEmail         string
+		requireHelmGAR   bool
+		requireOCIGAR    bool
+		requireCSR       bool
+		testKSAMigration bool
+		newRootSrcCfg    sourceConfig
+		newNSSrcCfg      sourceConfig
 	}{
 		{
-			name:          "Authenticate to Git repo on CSR with GKE WI",
-			fleetWITest:   false,
-			crossProject:  false,
-			sourcePackage: "hydration/kustomize-components",
-			sourceType:    v1beta1.GitSource,
-			gsaEmail:      gitproviders.CSRReaderEmail(),
-			rootCommitFn:  nomostest.RemoteRootRepoSha1Fn,
-			requireCSR:    true,
+			name:         "Authenticate to Git repo on CSR with GKE WI",
+			fleetWITest:  false,
+			crossProject: false,
+			rootSrcCfg:   sourceConfig{pkg: "hydration/kustomize-components", dir: "kustomize-components", commitFn: nomostest.RemoteRootRepoSha1Fn},
```

Comment:
nit: I expected `dir: "."` here. It looks like this is copying the nested directory over. Is that intentional? I feel this would be more intuitive if the contents were placed into the top level directory of the repo

Reply:
It is intentional. The default sync directory is not `.`, but `acme`:  https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/e2e/nomostest/gitproviders/repository.go#L61.

We could remove existing contents under `acme` and then copy all contents under `kustomize-components` to the `acme` directory. It feels like clearer to use a different sync directory.

Reply:
Ack, this seems reasonable